Library/Homebrew/formula_installer.rb

```diff
@@ -292,10 +292,21 @@ def effective_build_options_for(dependent, inherited_options=[])
 
   def inherited_options_for(dep)
     inherited_options = Options.new
+
+    # Universal is a special case of an inheritable option
     u = Option.new("universal")
     if (options.include?(u) || formula.require_universal_deps?) && !dep.build? && dep.to_formula.option_defined?(u)
       inherited_options << u
     end
+
+    # Handle other inheritable options
+    inheritableOptions = [Option.new("without-x11"), Option.new("without-x")]
+    inheritableOptions.each do |inheritableOption|
+      if options.include?(inheritableOption) && !dep.build? && dep.to_formula.option_defined?(inheritableOption)
```

Comment:
Question: Do we need this if statement?  There could be a case where A depends on B which depends on C.  A and C both have --without-x11 but B does not.  The if will cause C to be installed without --without-x11